### ISSUE TITLE
allOf not supported in multifiles spec

### ISSUE THREAD

**Issue by JFCote** [swagger-api/swagger-parser#604]
I have discovered that "allOf" seems to be unsupported when using multiple files.

If I have this simplified version of petstore.yaml in a single file, everything is ok:

swagger.yaml
```
swagger: '2.0'
info:
  version: '1.0.0'
  title: Swagger Petstore (Simple)
  description: A sample API that uses a petstore as an example to demonstrate features in the swagger-2.0 specification
  termsOfService: http://helloreverb.com/terms/
  contact:
    name: Swagger API team
    email: foo@example.com
    url: http://swagger.io
  license:
    name: MIT
    url: http://opensource.org/licenses/MIT
host: petstore.swagger.io
basePath: /api
schemes:
  - http
consumes:
  - application/json
produces:
  - application/json
paths:
  /pets:
    get:
      description: Returns all pets from the system that the user has access to
      operationId: findPets
      produces:
        - application/json
        - application/xml
        - text/xml
        - text/html
      parameters:
        - name: tags
          in: query
          description: tags to filter by
          required: false
          type: array
          items:
            type: string
          collectionFormat: csv
        - name: limit
          in: query
          description: maximum number of results to return
          required: false
          type: integer
          format: int32
      responses:
        '200':
          description: pet response
          schema:
            type: array
            items:
              $ref: '#/definitions/ShippingInfo'
definitions:
  PrintInfo:
    type: object
    description: "Print info of the shipping info"
    required:
      - B
    properties:
      B:
        type: boolean
  ShippingInfo:
    type: object
    description: "Information about the shipping for a device"
    allOf:
      - $ref: "#/definitions/PrintInfo"
      - type: object
        required:
          - A
        properties:
          A:
            type: integer
            format: int64
```

If I do the same thing, but using 2 files, `PrintInfo` is not in the definitions and `ShippingInfo` only contains the field `A` (should contain `A` and `B`)

swagger.yaml
```
swagger: '2.0'
info:
  version: '1.0.0'
  title: Swagger Petstore (Simple)
  description: A sample API that uses a petstore as an example to demonstrate features in the swagger-2.0 specification
  termsOfService: http://helloreverb.com/terms/
  contact:
    name: Swagger API team
    email: foo@example.com
    url: http://swagger.io
  license:
    name: MIT
    url: http://opensource.org/licenses/MIT
host: petstore.swagger.io
basePath: /api
schemes:
  - http
consumes:
  - application/json
produces:
  - application/json
paths:
  /pets:
    get:
      description: Returns all pets from the system that the user has access to
      operationId: findPets
      produces:
        - application/json
        - application/xml
        - text/xml
        - text/html
      parameters:
        - name: tags
          in: query
          description: tags to filter by
          required: false
          type: array
          items:
            type: string
          collectionFormat: csv
        - name: limit
          in: query
          description: maximum number of results to return
          required: false
          type: integer
          format: int32
      responses:
        '200':
          description: pet response
          schema:
            type: array
            items:
              $ref: './definitions.yaml#/definitions/ShippingInfo'
```

definitions.yaml
```
definitions:
  PrintInfo:
    type: object
    description: "Print info of the shipping info"
    required:
      - B
    properties:
      B:
        type: boolean
  ShippingInfo:
    type: object
    description: "Information about the shipping for a device"
    allOf:
      - $ref: "#/definitions/PrintInfo"
      - type: object
        required:
          - A
        properties:
          A:
            type: integer
            format: int64
```

The only difference between these 2 examples is the fact that one is in one file and the other is in 2 files.
@gracekarina This might interest you, it is highly related to this: https://github.com/swagger-api/swagger-parser/issues/581

Thanks

**Comment by gracekarina** [comment]
Hi @JFCote, this was also reported here #567
We are currently working on it ;)

**Comment by tzimisce012** [comment]
This problem comes from swagger parser project. I already fixed there, you just need to update the dependency.

**Comment by gracekarina** [comment]
Hi, I added the test and it passes 
https://github.com/swagger-api/swagger-parser/pull/618/commits/c10316753dd4c8519f6231486abbd4a386b20961

**Comment by JFCote** [comment]
I will try this today! Thanks for the update!

**Comment by JFCote** [comment]
@gracekarina I see it is fixed in the master but do you know when the version 1.0.34 will be released? Without a release, we can't update the dependency of swagger-codegen.

Thanks!

**Comment by gracekarina** [comment]
@JFCote Can you test it with current version?, I think that fix was made before

**Comment by JFCote** [comment]
@gracekarina Will do today and let you know.

**Comment by JFCote** [comment]
It seems to work and already be in the v2.4 of swagger-codegen! Thanks a lot @gracekarina

**Comment by JFCote** [comment]
@gracekarina Well, after some testing, I found 2 use cases that are not supported with this code. The one in the original post is supported and work great. But, I have found 2 slight variations that don't work. (I think)

I'm leaving work right now, I will do more testing and I will provide you with specific use cases (unit test).

Just let me know if you want them here or if you want me to create a new issue for this.
